src/rust/engine/process_execution/src/docker.rs

```diff
@@ -32,9 +33,14 @@ pub(crate) const SANDBOX_BASE_PATH_IN_CONTAINER: &str = "/pants-sandbox";
 pub(crate) const NAMED_CACHES_BASE_PATH_IN_CONTAINER: &str = "/pants-named-caches";
 pub(crate) const IMMUTABLE_INPUTS_BASE_PATH_IN_CONTAINER: &str = "/pants-immutable-inputs";
 
+/// Process-wide image pull cache.
+pub static IMAGE_PULL_CACHE: Lazy<ImagePullCache> = Lazy::new(ImagePullCache::new);
+
+/// Process-wide Docker connection.
+pub static DOCKER: Lazy<DockerOnceCell> = Lazy::new(DockerOnceCell::new);
```

Comment:
Making these global is probably fine, but note that elsewhere we accomplish this by `Arc`ing structures like these, and then passing references to them to each consumer: https://github.com/pantsbuild/pants/blob/7e94aefa3019456895db4232a73039efb9aed32b/src/rust/engine/src/context.rs#L486-L495

I would lightly prefer that for consistency, but I don't feel strongly about it.

Reply:
Both `ImagePullCache` and `DockerOnceCell` make use of interior mutability by storing an `Arc` internally. I can try re-adding references, but they shouldn't need an `Arc` to manage the whole struct.